src/services/completions.ts

```diff
@@ -1718,17 +1741,50 @@ namespace ts.Completions {
                 const symbolName = getNameForExportedSymbol(symbol, target);
                 const existingSuggestions = results.get(getSymbolId(original));
                 if (!some(existingSuggestions, s => s.symbolName === symbolName && moduleSymbolsAreDuplicateOrigins(moduleSymbol, s.origin.moduleSymbol))) {
-                    const origin: SymbolOriginInfoExport = { kind: SymbolOriginInfoKind.Export, moduleSymbol, isDefaultExport, isFromPackageJson };
+                    const origin: SymbolOriginInfoExport = {
+                        kind: SymbolOriginInfoKind.Export,
+                        moduleSymbol,
+                        isDefaultExport,
+                        isFromPackageJson,
+                        exportName,
+                        fileName: file?.fileName
+                    };
                     results.add(getSymbolId(original), {
                         symbol,
                         symbolName,
                         origin,
-                        skipFilter,
                     });
                 }
             }
         }
 
+        function getAutoImportSymbolFromCompletionEntryData(data: CompletionEntryData): { symbol: Symbol, origin: SymbolOriginInfoExport } | undefined {
+            const containingProgram = data.isPackageJsonImport ? host.getPackageJsonAutoImportProvider!()! : program;

```

Comment:
I think given that this is passed around we need to handle cases where data doesnt match what was passed around so instead of asserting we probably want to ensure that we handle if things are absent

Reply:
I don’t think that would make much difference—we already have an assertion later that `getCompletionDetails` can successfully find the completion by `entryId`. If a bad client passed an incorrect `name` and `source` combination, we would trigger that assertion. We could easily return `undefined` here, but an assertion would be triggered a little bit later when we couldn’t find the module in question. I could replace some of this with a `Debug.checkDefined` but I think it’s generally better to fail fast if `data` violates the assumptions we hold.